src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollectionItem.cs

```diff
@@ -19,23 +19,31 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
     /// </summary>
     public sealed partial class ZuneTrackCollectionItem : ObservableObject
     {
-        // Cache artist item values
-        // Keeping a minimum cache that updates with events allows us to
-        // avoid checking all items when a single item updates.
-        private ConcurrentDictionary<string, int> _lastKnownTrackArtistsCount = new();
-
         [ObservableProperty]
         private ITrackCollectionViewModel? _parentCollection;
 
         [ObservableProperty]
         private TrackViewModel? _track;
 
-        [ObservableProperty]
         private bool _shouldShowArtistList;
 
         [ObservableProperty]
         private ObservableCollection<MetadataItem> _artistNamesMetadata = new ObservableCollection<MetadataItem>();
 
+        /// <summary>
```

Comment:
We don't need to do this all manually anymore, now that we're just storing a simple bool value. Use the source generators via the `[ObservableProperty]` attribute, same way as the code above.